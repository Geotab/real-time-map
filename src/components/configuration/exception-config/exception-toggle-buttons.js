import React, { Component } from "react";
import { exceptionSearch } from "./exception-search";

export class ExceptionToggleButtons extends React.Component {
  constructor(props) {
    super(props);
    this.state = { visibility: true };
    this.toggleVisibility = this.toggleVisibility.bind(this);
  }

  toggleVisibility() {
    this.setState(state => {
      return { visibility: !state.visibility };
    });
  }

  render() {
    return (
      <div>
        <button
          id="clearExceptions"
          className={`toggleButton ${
            this.state.visibility ? "shown" : "notShown"
          }`}
          title="Toggle All Exceptions"
          onClick={() => {
            this.toggleVisibility();
            exceptionSearch.setVisibilityForAllItems(
              !this.state.visibility,
              this.props.setExceptionsDisplay
            );
          }}
        ></button>
        <button
          id="deleteExceptions"
          className="deleteButton"
          title="Delete All Exceptions"
          onClick={() =>
            exceptionSearch.deleteAllItems(this.props.setExceptionsDisplay)
          }
        ></button>
      </div>
    );
  }
}
