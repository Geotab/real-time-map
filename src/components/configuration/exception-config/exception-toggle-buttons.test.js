import React from "react";
import {ExceptionToggleButtons} from "./exception-toggle-buttons";
import { render, shallow } from "enzyme";

let wrapper;

let state;
beforeEach(() => {
  state = {
    visibility: true 
  };
});

test("should successfully render the component and match screenshot", () => {
  const component = render(<ExceptionToggleButtons />);
  expect(component).toMatchSnapshot();
});

test("should successfully togglenpm te component state", () => {
  wrapper = shallow(<ExceptionToggleButtons />);
  const instance = wrapper.instance();
  instance.toggleVisibility();
  expect(wrapper.state.visibility).toBeFalsy();
});