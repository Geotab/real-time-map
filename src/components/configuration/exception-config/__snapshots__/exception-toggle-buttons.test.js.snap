// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should successfully render the component and match screenshot 1`] = `
<div>
  <button
    class="toggleButton shown"
    id="clearExceptions"
    title="Toggle All Exceptions"
  />
  <button
    class="deleteButton"
    id="deleteExceptions"
    title="Delete All Exceptions"
  />
</div>
`;
